src/Core/Validation/FieldValidation/DateFieldValidator.php

```diff
@@ -2,29 +2,113 @@
 
 namespace SilverStripe\Core\Validation\FieldValidation;
 
+use Exception;
+use InvalidArgumentException;
 use SilverStripe\Core\Validation\FieldValidation\FieldValidator;
 use SilverStripe\Core\Validation\ValidationResult;
 
 /**
  * Validates that a value is a valid date, which means that it follows the equivalent formats:
  * - PHP date format Y-m-d
- * - SO format y-MM-dd i.e. DBDate::ISO_DATE
+ * - ISO format y-MM-dd i.e. DBDate::ISO_DATE
  *
  * Blank string values are allowed
  */
 class DateFieldValidator extends FieldValidator
 {
+    /**
+     * Minimum value as a date string
+     */
+    private ?string $minValue;
+
+    /**
+     * Minimum value as a unix timestamp
+     */
+    private ?int $minTimestamp;
+
+    /**
+     * Maximum value as a date string
+     */
+    private ?string $maxValue;
+
+    /**
+     * Maximum value as a unix timestamp
+     */
+    private ?int $maxTimestamp;
+
+    /**
+     * Converter to convert date strings to another format for display in error messages
+     *
+     * @var callable
```

Comment:
Not using strong typing because you cannot strongly type a callable property in PHP, even though you can assign it